### ISSUE TITLE
Remember state of maintenance:mode when running an upgrade

### ISSUE THREAD

**Issue by danimo** [owncloud/core#16429]
### Expected behavior

- `./occ maintenance:mode --on`
- Update packages
- Adjust things
- `./occ upgrade`
- Check if stuff did not explode
- `./occ maintenance:mode --off`

### Actual behavior

The upgrade mode disables maintenance mode when finished regardless on the previous state. Sure you can re-enable it, but that still leaves a few seconds where clients can hammer the server.

I've also read that there is a allow-only-admins mode. Using that for the verification would be even cooler, but I don't know how it interacts with the maintenance mode.

/cc @DeepDiver1975 @LukasReschke

**Comment by danimo** [comment]
Hmm, now when upgrading to 1.8.0-beta1 to 1.8.0-beta2, I get the following warning when trying to follow the procedure under "Expected behavior"

```
<warning>ownCloud is in maintenance mode</warning>
Maybe an upgrade is already in process. Please check the logfile (data/owncloud.log). If you want to re-run the upgrade procedure, remove the "maintenance mode" from config.php and call this script again.
```

Why is the maintenance mode (ab)used to indicate an upgrade?

**Comment by danimo** [comment]
Ah nevermind. As soon of any of the apps are upgrade candidates, it works.

**Comment by MorrisJobke** [comment]
> Using that for the verification would be even cooler, but I don't know how it interacts with the maintenance mode.

This is called singleuser mode: https://github.com/owncloud/core/blob/aae9274210ad80e3c313d55999684c03f6e06b2c/config/config.sample.php#L755-L759

**Comment by MorrisJobke** [comment]
Fix is in #16434
